File: broadcastreceiverlibrary/src/main/java/com/osapps/broadcastreceiverlibrary/network/NetworkObserver.java
package com.osapps.broadcastreceiverlibrary.network;


import com.osapps.broadcastreceiverlibrary.generic.BroadcastObserver;

/**
 * Created by osapps on 07/05/2018.
 */
public interface
NetworkObserver extends BroadcastObserver {
    void onNetworkChanged(NewNetworkStateObj networkStateObj);
    void registerReceivers(boolean register);
}

File: broadcastreceiverlibrary/src/main/java/com/osapps/broadcastreceiverlibrary/network/NetworkObserver.kt
package com.osapps.broadcastreceiverlibrary.network


import com.osapps.broadcastreceiverlibrary.generic.BroadcastObserver
import com.osapps.broadcastreceiverlibrary.headphones.HeadphonesObj

/**
 * Created by osapps on 07/05/2018.
 */
interface NetworkObserver : BroadcastObserver {
    fun onNetworkChanged(networkStateObj: NewNetworkStateObj)
}
